app/uk/gov/hmrc/soletraderidentificationfrontend/services/SubmissionService.scala

```diff
@@ -18,68 +18,171 @@ package uk.gov.hmrc.soletraderidentificationfrontend.services
+        JourneyCompleted(continueUrl)
+
+    case Right(true | false) =>
+      soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+        .map(_ => JourneyCompleted(continueUrl))
+
+    case Left(failure) =>
+      soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+        .map(_ => SoleTraderDetailsMismatch(failure))
+
+  }
+
+  private def handleSoleTraderJourneyWithBVCheck(journeyId: String,
+                                                 matchingResult: MatchingResult,
+                                                 journeyConfig: JourneyConfig,
+                                                 individualDetails: IndividualDetails)
+                                                (implicit hc: HeaderCarrier,
+                                                 ec: ExecutionContext): Future[SubmissionResponse] = matchingResult match {
+    case Right(true) if individualDetails.optSautr.nonEmpty =>
+      businessVerificationService.createBusinessVerificationJourney(journeyId, individualDetails.optSautr.getOrElse(throwASaUtrNotDefinedException)).flatMap {
+        case Right(BusinessVerificationJourneyCreated(businessVerificationUrl)) =>
+          Future.successful(StartBusinessVerification(businessVerificationUrl))
+        case _ =>
           for {
-            _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
             _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
           } yield
-            SoleTraderDetailsMismatch(failureReason)
+            JourneyCompleted(journeyConfig.continueUrl)
       }
-    } yield {
-      response
-    }
+
+    case Right(_) if noNino(individualDetails.optNino) => for {
+      _ <- createTrnService.createTrn(journeyId)
+      _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
+      _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+    } yield
+      JourneyCompleted(journeyConfig.continueUrl)
+
+    case Right(_) => for {
+      _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
+      _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+    } yield
+      JourneyCompleted(journeyConfig.continueUrl)
+
+    case Left(failureReason) =>
+      for {
+        _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
+        _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+      } yield
+        SoleTraderDetailsMismatch(failureReason)
+  }
+
+  private def throwASaUtrNotDefinedException: Nothing =
+    throw new IllegalStateException("This is a bug. sa-utr should be Defined but it is not!")
```

Comment:
I think we need a better error message along the lines of "Business verification requires a SA UTR to be defined".

Reply:
hey Richard that method is used in 2 places ..... but in one "Business verification requires a SA UTR to be defined" does not fit.... what should I do. .... in one place I leave it the way it is and in the other place I changed it???
Let me know .... leaving soon ;-)

Reply:
I'd pass the message in as an argument.